packages/react-components/src/components/ComplianceBanner/ComplianceBanner.tsx

```diff
@@ -71,15 +71,20 @@ export const _ComplianceBanner = (props: _ComplianceBannerProps): JSX.Element =>
     latestStringState: {
       callTranscribeState: 'off',
       callRecordState: 'off'
-    }
+    },
+    lastUpdated: Date.now()
   });
 
   // Only update cached props and variant if there is _some_ change in the latest props.
   // This ensures that state machine is only updated if there is an actual change in the props.
-  if (
+  const shouldUpdatedCached =
```

Comment:
nit: Name is a bit confusing and difficult to interpret.
Possibly rename to **shouldCacheBeUpdated** or **doesCacheNeedUpdate** 

(Another possible one is **isCacheUpdated** but this one adds it's own confusion as well when reading it's use case)